test/Microsoft.Framework.PackageManager.FunctionalTests/DnuSourcesTests.cs

```diff
@@ -0,0 +1,99 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.IO;
+using Microsoft.Framework.CommonTestUtils;
+using Microsoft.Framework.PackageManager.FunctionalTests;
+using Xunit;
+
+namespace Microsoft.Framework.PackageManager
+{
+    [Collection(nameof(PackageManagerFunctionalTestCollection))]
+    public class DnuSourcesTests
+    {
+        private readonly PackageManagerFunctionalTestFixture _fixture;
+
+        public DnuSourcesTests(PackageManagerFunctionalTestFixture fixture)
+        {
+            _fixture = fixture;
+        }
+
+        public static IEnumerable<object[]> RuntimeComponents
+        {
+            get
+            {
+                return TestUtils.GetRuntimeComponentsCombinations();
+            }
+        }
+
+        [Theory]
+        [MemberData(nameof(RuntimeComponents))]
+        public void DnuSources_ListsAllSources(string flavor, string os, string architecture)
```

Comment:
I'm not 100% happy with this test. It's a little too brittle. I might see if I can tidy it up before pushing.